Dynamo backend panics on `getTableStatus` in Teleport Cloud

I believe it was introduced in #27674. 


Bug details:
- Teleport version, master, head of v13 (13.2.3 not affected)
- Debug logs
```
{"caller":"runtime/panic.go:884","component":"dynamodb","level":"debug","message":"AWS session is created.","timestamp":"2023-07-25T18:46:29Z"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4fc3085]

goroutine 1 [running]:
github.com/gravitational/teleport/lib/backend/dynamo.(*Backend).getTableStatus(0xc0017ab000, {0x8e73978, 0xc00019a008}, {0xc00171ac80, 0x31})
	github.com/gravitational/teleport/lib/backend/dynamo/dynamodbbk.go:668 +0x1c5
github.com/gravitational/teleport/lib/backend/dynamo.New({0x8e73978?, 0xc00019a008}, 0xd14a9a0?)
	github.com/gravitational/teleport/lib/backend/dynamo/dynamodbbk.go:279 +0x86e
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthStorage(0xc0017908c0)
	github.com/gravitational/teleport/lib/service/service.go:5171 +0x26f
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthService(0xc0017908c0)
	github.com/gravitational/teleport/lib/service/service.go:1520 +0x4a
github.com/gravitational/teleport/lib/service.NewTeleport(0xc00178f000)
	github.com/gravitational/teleport/lib/service/service.go:1082 +0x2426
github.com/gravitational/teleport/e/tool/teleport/process.NewTeleport(0xc00178f000)
	github.com/gravitational/teleport/e/tool/teleport/process/process.go:57 +0x4df
github.com/gravitational/teleport/lib/service.Run({_, _}, {{0x7863e20, 0x2}, {0xc0001bbcc8, 0x11}, {0xc001716040, 0x1e}, {0x0, 0x0}, ...}, ...)
	github.com/gravitational/teleport/lib/service/service.go:682 +0xba
main.main()
	github.com/gravitational/teleport/e/tool/teleport/main.go:28 +0x225
```

--- a/lib/backend/dynamo/dynamodbbk.go
+++ b/lib/backend/dynamo/dynamodbbk.go
@@ -661,6 +661,13 @@ func (b *Backend) getTableStatus(ctx context.Context, tableName string) (tableSt
 			return tableStatusNeedsMigration, "", nil
 		}
 	}
+	// the billing mode can be empty unless it was specified on the
+	// initial create table request, and the default billing mode is
+	// PROVISIONED, if unspecified.
+	// https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BillingModeSummary.html
+	if td.Table.BillingModeSummary == nil {
+		return tableStatusOK, dynamodb.BillingModeProvisioned, nil
+	}
 	return tableStatusOK, aws.StringValue(td.Table.BillingModeSummary.BillingMode), nil
 }
 

